### PR TITLE
Proof generation performance tests

### PR DESCRIPTION
**Pull request by EvanJRichard**
## Summary
This PR proposes to add `rawTransaction.test.perf.ts`. It will benchmark the time it takes to call `raw.post` for a transaction with (1, 10, 100) notes.

```
[TEST RESULTS: Spends: 1, Outputs: 1]
Elapsed: 1,186.752 milliseconds

[TEST RESULTS: Spends: 10, Outputs: 1]
Elapsed: 7,944.69 milliseconds

[TEST RESULTS: Spends: 100, Outputs: 1]
Elapsed: 76,534.142 milliseconds

[TEST RESULTS: Spends: 1, Outputs: 1]
Elapsed: 1,108.69 milliseconds

[TEST RESULTS: Spends: 1, Outputs: 10]
Elapsed: 4,251.144 milliseconds

[TEST RESULTS: Spends: 1, Outputs: 100]
Elapsed: 35,963.016 milliseconds 
```

## Testing Plan
Self-explanatory: this PR solely adds tests. 

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
[X] No
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by EvanJRichard** [comment]
Mangled the history, so force-pushed to a good state. Turning on auto-merge since I may not be around to click Merge after review.

File: ironfish/src/primitives/rawTransaction.test.perf.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    amount: totalAmount - fee,
+    fee: fee,
+    expiration: inputCount + 2,
+    burns: [],
+    mints: [],
+  })
+  const start = BenchUtils.start()
+  const posted = raw.post(account.spendingKey)
+  const end = BenchUtils.end(start)
+  expect(posted).toBeTruthy()
```
Comment (NullSoldier) [inline ironfish/src/primitives/rawTransaction.test.perf.ts:84]: this expect can never fail

File: ironfish/src/primitives/transaction.test.perf.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    // Spends
+    for (const i of TEST_AMOUNTS) {
+      const results = await runTest(account, i, 1)
+      printResults(results)
+    }
+
+    // Outputs
+    for (const i of TEST_AMOUNTS) {
```
Comment (NullSoldier) [inline ironfish/src/primitives/transaction.test.perf.ts:47]: the first test is duplicated for outputs. Should we just make TEST_AMMOUNTS a tuple?

**Review by NullSoldier — APPROVED** [review]

File: ironfish/src/primitives/transaction.test.perf.ts (hunk 2 above)
Comment (mat-if) [reply ironfish/src/primitives/transaction.test.perf.ts]: good call, made it more explicit by making TEST_AMOUNTS an array of { spends, outputs } which is easier to read and removes the duplicate

**Pull request merged**